### DISCUSSION TITLE
Mutiple UEs Multiple gNBs on ZMQ and GNR setup

### DISCUSSION THREAD

**Discussion by goyban** [srsran/srsRAN_Project#828] [General Help] [answered]
Hello everyone!

I want to setup 3 gNBs, and multiple UEs using ZMQ and GNR (topology provided below):

![image](https://github.com/user-attachments/assets/bfce73cf-7b81-460e-802b-9c4727c2c480)

So far I got my answers about gNB <-> O5GS part by reading this [discussion](https://github.com/srsran/srsRAN_Project/discussions/561).
What I'm not sure about is how to configure GNR and ZMQ ports (UE <-> gNB) for the second and third gNB.
Is the configuration below correct?

1st gNB (Basically same as this [tutorial](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#multi-ue-emulation)) :

```
| Port Direction | gNB1  | UE1  | UE2  | UE3  |
| TX             | 2000  | 2101 | 2201 | 2301 |
| RX             | 2001  | 2100 | 2200 | 2300 |
```

2nd gNB (Should be another flow graph?):

```
| Port Direction | gNB2  | UE4  | UE5  |
| TX             | 3000  | 3101 | 3201 |
| RX             | 3001  | 3100 | 3200 |
```

3rd gNB (No Broker needed?):

```
| Port Direction | gNB3  | UE6  |
| TX             | 4000  | 4001 |
| RX             | 4001  | 4000 |
```

Thanks in advance!

**Comment by pgawlowicz** [comment, marked as answer]
The ports look OK. 
It is important to connect them correctly so that
- the DL signal goes from gnb to gnuradio, where it is copied and send to all UEs,
- and the UL signal goes from UEs, to gnuradio where it is aggregated and sent to the gnb.

**Reply by goyban** [reply]
Thank you so much for your answer, is it correct that the third gNB doesn't need GNR since it's only 1 UE?

**Reply by pgawlowicz** [reply]
correct, you can connect it directly
